request/quic_transport.go

```diff
@@ -1,26 +1,34 @@
 package request
 
 import (
+	"log"
 	"net/http"
 	"strings"
+	"sync"
 	"sync/atomic"
 
 	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
 	"github.com/quic-go/quic-go/http3"
 )
 
 // QuicTransport is modified/enhanced RoundTripper.
 // Thread safe.
 type QuicTransport struct {
-	inner    *atomic.Pointer[http3.RoundTripper]
-	createFn func() *http3.RoundTripper
+	// for protects the access to inner and to shouldRecreate.
+	// This will allow multiple requests in the same time for inner, but only one can recreate it
+	mu sync.RWMutex
+	// when executing the requests set this to true on error and when inner is recreated is set to false.
+	// This is used to ensure only one failed request will recreate inner value on error
+	shouldRecreate atomic.Bool
```

Comment:
if you think that having also this makes too complicated the logic, then it can be removed. 
This only purpose is to prevent recreation of the `inner` multiple times, if more requests fail in the same time.